dhctl/pkg/preflight/check_staticinstances_with_sshcredentials.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2025 Flant JSC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package preflight
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"strings"
+
+	"golang.org/x/crypto/ssh"
+	"gopkg.in/yaml.v3"
+
+	"github.com/deckhouse/deckhouse/dhctl/pkg/util/input"
+)
+
+type SSHCredential struct {
+	User         string
+	Key          string
+	SudoPassword string
+	Port         string
+}
+
+func (pc *Checker) CheckStaticInstancesSSH(ctx context.Context) error {
+	config := input.YAMLSplitRegexp.Split(pc.metaConfig.ResourcesYAML, -1)
+
+	for _, doc := range config {
+
+		var res map[string]interface{}
+		if err := yaml.Unmarshal([]byte(doc), &res); err != nil {
+			return fmt.Errorf("cannot unmarshal YAML: %v", err)
+		}
+
+		if res["kind"] != "StaticInstance" {
+			continue
+		}
+
+		meta := res["metadata"].(map[string]interface{})
+		instanceName := meta["name"].(string)
+
+		spec := res["spec"].(map[string]interface{})
+		address := spec["address"].(string)
+
+		credRef := spec["credentialsRef"].(map[string]interface{})
+		credName := credRef["name"].(string)
+
+		cred, err := findSSHCredentials(config, credName)
+		if err != nil {
+			return fmt.Errorf("instance %s: %v", instanceName, err)
+		}
+
+		if err := testSSHConnection(address, cred.Port, cred); err != nil {
+			return fmt.Errorf("cannot connect to %s (%s:%s): %v", instanceName, address, cred.Port, err)
+		}
+	}
+
+	return nil
+}
+
+func findSSHCredentials(docs []string, name string) (*SSHCredential, error) {
```

Comment:
we can first find credentials, returns map name -> credentials and find creds by name in map

Reply:
My reason. 

We unmarshal every document. if staticinstance found we pass all documents to another function and parse all documents again for every static instance.

I know, that commander create a lot additional resources to config before bootstrap. It will heavy to execution.

But in my variant we have one cycle when we collect list of static instances and map StaticInstance.Name -> SSHCredentials (during parse validate it one time)

and for for list by staticInstance -> check that for staticInstance in map (have SSHCredentials) and use SSHCreds for checking